python/lsst/ts/m2/csc.py

```diff
@@ -67,326 +105,640 @@ def __init__(
+
+        elif message_name == "tangentActuatorSteps":
+            self.tel_tangentActuatorSteps.set_put(steps=message["steps"])
+
+        elif message_name == "axialEncoderPositions":
+            self.tel_axialEncoderPositions.set_put(position=message["position"])
+
+        elif message_name == "tangentEncoderPositions":
+            self.tel_tangentEncoderPositions.set_put(position=message["position"])
+
+        elif message_name == "ilcData":
+            self.tel_ilcData.set_put(status=message["status"])
+
+        elif message_name == "displacementSensors":
+            self.tel_displacementSensors.set_put(
+                thetaZ=message["thetaZ"], deltaZ=message["deltaZ"]
+            )
+
+        elif message_name == "forceBalance":
+            self.tel_forceBalance.set_put(
+                fx=message["fx"],
+                fy=message["fy"],
+                fz=message["fz"],
+                mx=message["mx"],
+                my=message["my"],
+                mz=message["mz"],
             )
-            await self._run_task_in_new_event_loop(
-                self.model.handle_position_mirror, mirror_position_set_point
+
+        elif message_name == "netForcesTotal":
+            self.tel_netForcesTotal.set_put(
+                fx=message["fx"],
+                fy=message["fy"],
+                fz=message["fz"],
             )
 
-            self.evt_m2AssemblyInPosition.set_put(inPosition=True)
+        elif message_name == "netMomentsTotal":
+            self.tel_netMomentsTotal.set_put(
+                mx=message["mx"],
+                my=message["my"],
+                mz=message["mz"],
+            )
 
-        except RuntimeError:
-            self.fault(
-                code=POSITION_MIRROR_ERROR,
-                report="Failed to position the mirror.",
-                traceback="",
+        elif message_name == "powerStatus":
+            self.tel_powerStatus.set_put(
+                motorVoltage=message["motorVoltage"],
+                motorCurrent=message["motorCurrent"],
+                commVoltage=message["commVoltage"],
+                commCurrent=message["commCurrent"],
             )
 
-    async def _run_task_in_new_event_loop(self, func, *args):
-        """Run the task in the new event loop.
+        else:
+            self.log.info(f"Receive the unexpected telemetry: {message}")
+
+    async def connect_server(self):
+        """Connect the TCP/IP server."""
+
+        # self.simulation_mode is the attribute from upstream: BaseCsc
+        if self.simulation_mode == 0:
+            host = self._host
+            port_command = self._port_command
+            port_telemetry = self._port_telemetry
+
+        else:
+            host = tcpip.LOCAL_HOST
+            port_command = self._mock_server.server_command.port
+            port_telemetry = self._mock_server.server_telemetry.port
+
+        self.model.start(host, port_command, port_telemetry)
```

Comment:
I think the confusion here is the implementation of connection task. In the `model.start()`, the async task of connection allows the re-establish of connection if it breaks. Therefore, it might not be a good idea of `await` to wait for the connection is established. In addition, I removed the following `async.sleep()` to avoid the confusion in the commit: 9354e3d. Thanks!